src/ServiceManagement/Common/Commands.ScenarioTest/Resources/ServiceManagement/ServiceManagementTests.ps1

```diff
@@ -296,6 +296,73 @@ function Run-AutoGeneratedVirtualMachineCmdletTests
     }
 }
 
+# Run dSMS Hosted Service test
+function Run-DSMSHostedServiceTest
+{
+	# Setup
+    $svcName = 'pstest' + (Get-CloudServiceName);
+	$location = Get-DefaultLocation;
+
+    $storageName = 'pstest' + (getAssetName);
+    New-AzureStorageAccount -StorageAccountName $storageName -Location $location;
+
+    # Associate the new storage account with the current subscription
+    Set-CurrentStorageAccountName $storageName;
+
+	$testMode = Get-ComputeTestMode;
+    if ($testMode.ToLower() -ne 'playback')
+    {
+        $cspkg = '.\Resources\ServiceManagement\Files\dSMSTest.cspkg';
+    }
+    else
+    {
+        $cspkg = "https://${storageName}.blob.azure.windows.net/blob/dSMSTest.cspkg";
+    }
+    $cscfg = "$TestOutputRoot\Resources\ServiceManagement\Files\dSMSTest.cscfg";
+	$cscfgChanged = "$TestOutputRoot\Resources\ServiceManagement\Files\dSMSTest-changed.cscfg";
+	
+	# Create a temporary self-signed cert
+	$cert = New-SelfSignedCertificate -DnsName "example.local" -CertStoreLocation "Cert:\CurrentUser\My";
+	$certPath = "Cert:\CurrentUser\My\$($cert.Thumbprint)"
+	# Update the cscfg to use the new cert
+	(Get-Content $cscfg) | ForEach-Object {$_ -replace "\{\{PLACEHOLDER\}\}", $cert.Thumbprint} | Set-Content $cscfgChanged;
+
+    try
+    {
+		# Create Hosted Service
+		$result = New-AzureService -ServiceName $svcName -Location $location -Label $svcName -Description $svcName;
+
+		# Upload the certificate
+		Add-AzureCertificate -ServiceName $svcName -CertToDeploy $cert;
+
+        # Deploy to staging
+        $result = New-AzureDeployment -ServiceName $svcName -Package $cspkg -Configuration $cscfgChanged -Label $svcName -Slot Staging;
+
+		# Get Deployment
+		$deploy = Get-AzureDeployment -ServiceName $svcName -Slot Staging;
```

Comment:
So before the change, if the you have a line in cscfg like: "<Certificate name="dSMSCert" sourceLocation="/genevacustomers/monitoring/prod/warmpath/test/auxweb/certificates/chained/portaldogfoodmdscert.pfx" />", then you would get null ref error when you create a new deployment. Is my understanding correct?

Reply:
Yes. If you didn't supply a thumbprint attribute, it would NullRef when you perform Get-AzureDeployment

Reply:
Then shouldn't the NullRef exception be thrown when you do New-AzureDeployment? The PR topic says the exception happens when you do Get-AzureDeployment

Reply:
The NullRef was only thrown when I ran Get-AzureDeployment. The test goes through CRUD to ensure it's not occurring anywhere else. I'd rather have extra coverage :)